libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
-Beats 6.0. We recommend using this file as a reference only. It's not intended
-to be used in production.
+[source,shell]
+----
+metricbeat setup --template
+----
 
-==== Dashboard upgrades
+If your {beats} configuration use an output other than `elasticsearch`,
+temporarily enable the `elasticsearch` output when you run the `setup` command.
+For example:
 
-We recommend re-importing the Kibana dashboards after the Beats and Kibana
-upgrades are complete. This way, you will take advantage of the new Beats
-dashboards created for the 6.0 release.
+[source,shell]
+--
+metricbeat setup --template -E output.logstash.enabled=false -E 'output.elasticsearch.hosts=["localhost:9200"]'
+--
 
-See <<breaking-changes-import-dashboards>> in the Breaking Changes
-documentation.
+[[non-es-outputs]]
+==== How to use versioned templates when the output is not {es}
 
-NOTE: If you've modified the Beats dashboards manually, reimporting them will
-overwrite your changes. Consider saving your dashboards under new IDs, so
-that the reimport doesn't overwrite them.
+If you've configured {beats} to send events to a different output, such as {ls},
+make sure you use versioned templates and indices. Otherwise, after you
+upgrade, there will be field conflicts. To use versioned index templates and
+indices, use the metadata from {beats} to set the index, and allow {beats} to
+manage the template.
 
-==== Heartbeat package names
+For example, for {ls}, set the {ls} `index` setting as recommended in the
+{logstash-ref}/plugins-inputs-beats.html[{beats} input plugin] documentation and
+allow {beats} to manage the index template:
 
-The DEB and RPM packages for Heartbeat are now named `heartbeat-elastic` to
-avoid conflicts with a different `heartbeat` project.  The `.deb` and `.rpm`
-file names haven't changed, just the package name in the repositories.
+[source,json]
+----
+    manage_template => false
+    index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}" 
+----
 
-[[upgrade-mapping-template]]
-=== Upgrade the Elasticsearch mapping template
+When you use this configuration, the index name is set to match the index
+pattern in the {beats} index template.
+
+IMPORTANT: The index name must match the index pattern in the {beats} index
+template. For example, if {ls} sends events to an index called
++metricbeat-7-{localdate}+, but the index template expects indices to match
++metricbeat-{stack-version}-*+, you may encounter mapping errors and be unable
+to index {beats} events.
 
-Starting with Beats 6.0, the mapping templates and the default index names are
-versioned. For example, Metricbeat 6.0.0 creates indices like this:
+==== Upgrade dashboards
+
+We recommend that you re-import the {kib} dashboards after upgrading {kib} and
+{beats}. This way, you can take advantage of the new {beats} dashboards created
+for the 7.0 release. 
+
+Re-importing the dashboards will overwrite any manual changes that you've
```

Comment:
This is not correct. All 7.0 dashboards have an `-ecs` postfix so not 6.x dashboards will be overwritten. The two versions can coexist.